wcfsetup/install/files/lib/system/worker/UserRebuildDataWorker.class.php

```diff
@@ -289,48 +288,19 @@ public function execute()
                 UserStorageHandler::getInstance()->reset($resetAvatarCache, 'avatar');
             }
 
-            // Create WebP variants of existing cover photos.
-            $userProfiles = new UserProfileList();
+            // Migrate old cover photos into the new file system.
+            $userProfiles = new UserList();
             $userProfiles->getConditionBuilder()->add("user_table.userID IN (?)", [$userIDs]);
             $userProfiles->getConditionBuilder()->add("user_table.coverPhotoHash IS NOT NULL");
-            $userProfiles->getConditionBuilder()->add("user_table.coverPhotoHasWebP = ?", [0]);
             $userProfiles->readObjects();
-            foreach ($userProfiles as $userProfile) {
-                $editor = new UserEditor($userProfile->getDecoratedObject());
-                $coverPhoto = $userProfile->getCoverPhoto(true);
-                if ($coverPhoto instanceof DefaultUserCoverPhoto) {
-                    // The default cover photo can be returned if the user has a
-                    // cover photo, but it has been disabled by an administrator.
-                    continue;
-                }
-
-                // If neither the regular, nor the WebP variant is readable then the
-                // cover photo is missing and we must clear the database information.
-                if (
-                    !\is_readable($coverPhoto->getLocation(false))
-                    && !\is_readable($coverPhoto->getLocation(true))
-                ) {
-                    $editor->update([
-                        'coverPhotoHash' => null,
-                        'coverPhotoExtension' => '',
-                    ]);
-
-                    continue;
-                }
-
-                if ($coverPhoto instanceof IWebpUserCoverPhoto) {
-                    $result = $coverPhoto->createWebpVariant();
-                    if ($result !== null) {
-                        $data['coverPhotoHasWebP'] = 1;
-
-                        // A fallback jpeg image was just created.
-                        if ($result === false) {
-                            $data['coverPhotoExtension'] = 'jpg';
-                        }
-
-                        $editor->update($data);
-                    }
-                }
+            foreach ($userProfiles as $user) {
+                $file = FileEditor::createFromExistingFile(
+                    SetCoverPhoto::getCoverPhotoLocation($user, false),
```

Comment:
This is an odd way of doing it, commands should not have publicly callable methods. If you need a helper method for the legacy location then add those to the `UserCoverPhoto` class as `getLegacyLocation(…)`.